Write ";" after static functions

aschoenebeck opened the pull request:
JSHint complained about missing ";" after function declarations in
generated code, which is correct.

Before it was:

``` javascript
Foo.Bar = function() {
}
```

Now public static functions and property accessors are generated as
assignment with trailing ";":

``` javascript
Foo.Bar = function() {
};
```

aschoenebeck committed:
Write ";" after static functions

JSHint complained about missing ";" after function declarations in
generated code, which is correct.

Before it was:

Foo.Bar = function() {
}

Now public static functions and property accessors are generated as
assignment with trailing ";":

Foo.Bar = function() {
};

theoutlander commented:
Aren't ';' optional in JS?

nikhilk commented:
I suppose JSLint type of tools are strict, and in this case the scenario generates an assignment statement rather than a function declaration statement, so to be precise a ';' is called for. I would expect the ';' to be stripped out by the minimizer, but that is something to verify.

Also, it would be good if the unit tests can be updated, as this will change the baselines. You can run the unit tests by opening up the test solution and just using the VS UI to run all tests. The changed generated scripts be generated into the TestResults folder, making it easy to update the baselines that do need updates.

aschoenebeck commented:
I'm trying to fix the tests with _VS 2012_. Some tests fail not because of ";" but because of the different order of class definitions. For example in ExpressionTests.TestBinary.txt the App class is defined before Data and the test will fail. Should I just add ";" and ignore the ordering?

nikhilk commented:
Hmmm... thats pretty strange. The compiler should be generating script in the order of the types as defined in the sources. Is that the only test showing that? If so, ignore that test case, and I'll look into it.

aschoenebeck committed:
Update tests for write ";" after static functions

Changing compiler output requires updated tests. Added ";" in all
necessary places.

aschoenebeck commented:
Could you check if all tests pass now? I still have 32 failing tests because of type reordering. Script# and Tests compiled with VS 2012 Express.
